src/models/losses.py

```diff
@@ -141,7 +140,7 @@ def damsm_loss(
         mask[i] = 0
         masks.append(mask.reshape(1, -1))  # shape: (1, batch)
 
-        numb_words = cap_len_list[i]
+        numb_words = int(cap_lens[i])
```

Comment:
Do you even need to:

1. Cast it?
2. Make it a separate variable?

To my view, you can just do `query_words = words_emb[i, :, :cap_lens[i]].unsqueeze(0)`